### PR TITLE
Cat 1496 women yoi recat it

### PR DESCRIPTION
**Pull request by nnehabharti**

### CODE REVIEW

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy [hunk 1]
```diff
@@ -512,6 +593,18 @@ class Elite2Api extends WireMockRule {
         .withStatus(200)))
   }
 
+  void stubUncategorisedForWomenYO(bookingId, location) {
```
Comment (Olusola-Opencast) [inline integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:596]: nit picking can you change the name to stubUncategorisedForWomenYOI?  Just makes it clearer.

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy [hunk 2]
```diff
@@ -766,6 +872,76 @@ class Elite2Api extends WireMockRule {
         .withStatus(200)))
   }
 
+
+  def stubGetOffenderDetailsWomenYO(int bookingId, offenderNo = 'C0001AA', youngOffender = true, indeterminateSentence = false, category, multipleSentences = false, nextReviewDate = '2020-01-16') {
```
Comment (Olusola-Opencast) [inline integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:876]: nit pick can you change this to stubGetOffenderDetailsWomenYOI

File: server/views/formPages/supervisor/recatReview.html [hunk 3]
```diff
@@ -139,6 +139,9 @@ <h1 class="govuk-heading-l openConditionsHeader">Open Conditions</h1>
     {% endif %}
 
     {% set overriddenCategoryTextHtml %}
+      {% if (currentCat == "R") %}
```
Comment (Olusola-Opencast) [inline server/views/formPages/supervisor/recatReview.html:142]: @kberec this was added by Neha because she found a bug that needed a quick fix.

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy (hunk 1 above)
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:596]: ok will update it

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/DashboardWomenSpecification.groovy [hunk 4]
```diff
@@ -162,6 +162,104 @@ class DashboardWomenSpecification extends AbstractSpecification {
       'Total 8']
   }
 
+
+  def "The initial cat dashboard should show correct women YOI stats data"() {
```
Comment (kberec) [inline integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/DashboardWomenSpecification.groovy:166]: Do we need additional test scenarios like this one "The initial cat dashboard should show correct women YOI stats data" and "The recat dashboard should show correct women YOI stats data"?
The more tests we add here the slower it gets to run the full suite.
Other tests e.g. "The recat dashboard should show correct change table for Women"  have been now extended with YOI categories, is that sufficient?

File: test/routes/form.test.js [hunk 5]
```diff
@@ -745,7 +745,6 @@ describe('GET /supervisor/review', () => {
         expect(res.text).not.toContain('overriddenCategoryT')
         expect(res.text).not.toContain('overriddenCategoryJ')
         expect(res.text).toContain('overriddenCategoryR')
-        expect(res.text).not.toContain('value="R" checked')
```
Comment (kberec) [inline test/routes/form.test.js:748]: Why was this changed?

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy [hunk 6]
```diff
@@ -75,4 +77,57 @@ class DecisionSpecification extends AbstractSpecification {
     responseAfterClosed.ratings == [decision: [category: "R"]]
     responseAfterClosed.openConditionsRequested == false
   }
+
+//  Young Offender Category decision page tests
```
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy:81]: No , I have uncommented it and updated the tests. I have written it before you pushed your changes so forgot to uncomment and run it and test.

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy [hunk 7]
```diff
@@ -686,7 +792,7 @@ class Elite2Api extends WireMockRule {
         .withStatus(200)))
   }
 
-  def stubGetOffenderDetailsWomen(int bookingId, offenderNo = 'ON700', youngOffender = false, indeterminateSentence = false, category = 'U(Unsentenced)', multipleSentences = false, nextReviewDate = '2020-01-16') {
+  def stubGetOffenderDetailsWomen(int bookingId, offenderNo = 'ON700', youngOffender = false, indeterminateSentence = false, category, multipleSentences = false, nextReviewDate = '2020-01-16') {
```
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:795]: I use this method in recat too so I update  the value of  category according to the requirement

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy [hunk 8]
```diff
@@ -68,7 +68,7 @@ class DatabaseUtils {
     doCreateData(-1, bookingId, status, json)
   }
 
-  def createDataWithStatusWomen(id, bookingId, status, json, userId, prisonId) {
+  def createDataWithStatusWomen(id, bookingId, status, json,userid,prison) {
```
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy:71]: Yes, they are being used in Supervisor test to create db for women

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy (hunk 7 above)
[the remark this answers is not in the document]
Comment (kberec) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:795]: Yes, but I think recommendation is to have optional parameters as the trailing ones. E.g. https://stackoverflow.com/questions/70966722/default-value-parameter-in-groovy-as-middle-argument
In this example, bookingId is mandatory, all the rest should be optional, i.e. with default values.

File: test/routes/form.test.js (hunk 5 above)
Comment (nnehabharti) [reply test/routes/form.test.js:748]: I dont touch it tbh. is there something i need to update in it before I push the commented changes?

Comment (kberec) [reply test/routes/form.test.js:748]: I'm not sure why this is showing as removal. On main there is a line: `expect(res.text).not.toContain('value="R" checked')` Maybe something has gone wrong with merging the main back in? Any ideas @Olusola-Opencast ?

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy (hunk 8 above)
[the remark this answers is not in the document]
Comment (kberec) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy:71]: Seems to me that parameters userid and prison are not used.

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy (hunk 6 above)
[the remark this answers is not in the document]
Comment (kberec) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy:81]: Let me know when it is pushed @nnehabharti

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy (hunk 7 above)
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:795]: I have passed the value as category = 'U(Unsentenced)'? where ever its needed and category = 'R' where its not needed to be U

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy (hunk 8 above)
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/model/DatabaseUtils.groovy]: It looks like that but I did pass the values otherwise it used to give me 404

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy (hunk 6 above)
[the remark this answers is not in the document]
Comment (nnehabharti) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/specs/ratings/DecisionSpecification.groovy]: I have pushed the code now @kberec

**Review by kberec — APPROVED** [review]

File: integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy (hunk 7 above)
[the remark this answers is not in the document]
Comment (kberec) [reply integration-tests/src/test/groovy/uk/gov/justice/digital/hmpps/cattool/mockapis/Elite2Api.groovy:795]: We'll discuss at some other point
